File: barberia/src/Services/RegisterService.js
import axios from "axios";

const BASE_API_URL = 'http://localhost:8080';
localStorage.getItem("id");
class RegisterService{
    addShop(register){
        return axios.post(BASE_API_URL+'/api/v1/addShop',register);
    }

    viewProfile(){
        return axios.get(BASE_API_URL,'/viewProfile');
    }

    shopLogin(data){
        return axios.post(BASE_API_URL+'/api/v1/shoplogin',data)
    }


    getRegisterShopById(shopno){
        return axios.get(BASE_API_URL+'/api/v1/getShopById/'+shopno)
    }

    getAllRegisterShop(){
        return axios.get(BASE_API_URL+'/api/v1/allShop')
    }

    getShopByEmail(email){
        return axios.get(BASE_API_URL+'/api/v1/findShopByEmail/'+email)
    }

    updateShopProfile(profile){
        return axios.put(BASE_API_URL+'/api/v1/updateShopProfile/{shopno}',profile)
    }

    placeOrder(order){
        return axios.post(BASE_API_URL+'/api/v4/addOrder',order)
    }

    getAllOrderBySr(shopno){
        return axios.get(BASE_API_URL+'/api/v4/ordersr/'+shopno)
    }
    getAllOrderByUr(userid){
        return axios.get(BASE_API_URL+'/api/v4/orderur/'+userid)
    }
}

    
export default new RegisterService();

File: Barberia_BackEnd/src/main/java/com/barberia/service/RegisterService.java
package com.barberia.service;

import java.util.List;

import com.barberia.model.RegisterShop;

public interface RegisterService {
	
	public RegisterShop addShop(RegisterShop registerShop);
	
	public List<RegisterShop> viewProfile();
	
	public RegisterShop getShopById(int shopno);
	
	public RegisterShop updateShopProfile(RegisterShop registerShop);
	
	public RegisterShop login(RegisterShop registerShop);
	
 
	public RegisterShop findByEmail(String email);
}
